### PR TITLE
ui: disable toggles while onroad

### LINKED ISSUE #20965: UI: disable toggles while onroad

### PR DESCRIPTION
**Pull request by deanlee**
resolve https://github.com/commaai/openpilot/issues/20965

### CODE REVIEW

File: selfdrive/ui/qt/offroad/settings.cc [hunk 1]
```diff
@@ -25,69 +25,95 @@
...
+      true,
+    });
   }
 
-  bool record_lock = params.getBool("RecordFrontLock");
-  record_toggle->setEnabled(!record_lock);
+  for (auto &[param, title, desc, icon, enabled] : toggles) {
+    auto toggle = new ParamControl(param, title, desc, icon, this);
+    toggle->setEnabled(enabled);
+    if (enabled) {
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/offroad/settings.cc:112]: instead of this, let's check `param + "Lock"`

File: selfdrive/ui/qt/offroad/settings.h [hunk 2]
```diff
@@ -12,6 +12,7 @@
 #include "selfdrive/ui/qt/widgets/controls.h"
 
 // ********** settings window + top-level panels **********
+class SettingsWindow;
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/offroad/settings.h:15]: move settings window up instead?

**Comment by incognitojam** [comment]
Hey, what was the reasoning behind this change?

**Comment by adeebshihadeh** [comment]
Flipping the toggles while onroad has no effect until the next drive, so this was to reduce confusion around that.

**Pull request merged**